assets/js/util.js

```diff
@@ -242,8 +243,21 @@ function synchronizeTextareaHeights() {
     $('#translatedText').css('height', originalTextScrollHeight + 'px');
 }
 
-/*:: export {synchronizeTextareaHeights, modeEnabled, ajaxSend, ajaxComplete, filterLangList, onlyUnique,
-    SPACE_KEY_CODE, ENTER_KEY_CODE, HTTP_OK_CODE, HTTP_BAD_REQUEST_CODE, XHR_LOADING, XHR_DONE} */
+function callApy(options, endpoint, requestName) {
+    var callApyObject = Object.assign({
+        url: config.APY_URL + endpoint,
+        beforeSend: ajaxSend,
+        contentType: 'application/x-www-form-urlencoded; charset=UTF-8',
+        dataType: 'json'
+    }, options);
+
+    callApyObject.type = (callApyObject.data.q.length > THRESHOLD_REQUEST_LENGTH) ? 'POST' : 'GET';
+    window[requestName] = $.ajax(callApyObject);
```

Comment:
Actually, can we do something like this:

```
 +  return $.ajax(Object.assign({
 +        url: config.APY_URL + endpoint,
 +        beforeSend: ajaxSend,
 +        contentType: 'application/x-www-form-urlencoded; charset=UTF-8',
 +        dataType: 'json',
 +        type: (callApyObject.data.q.length > THRESHOLD_REQUEST_LENGTH) ? 'POST' : 'GET',
 +    }, options));
```
and then handle the `requestName = ...` at the callsite?